internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -463,6 +465,28 @@ func (r *AgentServiceConfigReconciler) newPostgresSecret(instance *aiv1beta1.Age
 }
 
 func (r *AgentServiceConfigReconciler) newAssistedServiceDeployment(instance *aiv1beta1.AgentServiceConfig, serviceURL *url.URL) (*appsv1.Deployment, controllerutil.MutateFn) {
+
+	// TODO(djzager): I think we want to use a hash or the configmap's
+	// resource version to force deployment to restart if configmap is
+	// changed. Alternatively, since this isn't "supported" the expectation
+	// could be that you do this yourself by deleting the assisted-service
+	// pod when you change the configmap.
```

Comment:
I agree that people using this annotation should be responsible for restarting the pod and everything related/needed to use it. Adding any automation around it will set the wrong expectations.

Reply:
Agreed